src/surfaceqg.jl

```diff
@@ -0,0 +1,370 @@
+module SurfaceQG
+
+export
+  Problem,
+  set_b!,
+  updatevars!,
+
+  kinetic_energy,
+  buoyancy_variance,
+  buoyancy_dissipation,
+  buoyancy_work,
+  buoyancy_drag,
+  buoyancy_advection,
+  kinetic_energy_advection
+
+  using
+    CUDA,
+    Reexport
+
+@reexport using FourierFlows
+
+using LinearAlgebra: mul!, ldiv!
+using FourierFlows: parsevalsum
+
+nothingfunction(args...) = nothing
+
+"""
+    Problem(; parameters...)
+
+Construct a Surface QG turbulence problem.
+"""
+function Problem(dev::Device=CPU();
+  # Numerical parameters
+          nx = 256,
+          Lx = 2π,
+          ny = nx,
+          Ly = Lx,
+          dt = 0.01,
+  # Drag and/or hyper-/hypo-viscosity
+           ν = 0,
+          nν = 1,
+           μ = 0,
+          nμ = 0,
+  # Timestepper and equation options
+     stepper = "RK4",
+       calcF = nothingfunction,
+  stochastic = false,
+           T = Float64)
+
+  grid = TwoDGrid(dev, nx, Lx, ny, Ly; T=T)
+
+  params = Params{T}(ν, nν, μ, nμ, calcF)
+
+  vars = calcF == nothingfunction ? Vars(dev, grid) : (stochastic ? StochasticForcedVars(dev, grid) : ForcedVars(dev, grid))
+
+  equation = Equation(params, grid)
+
+  return FourierFlows.Problem(equation, stepper, dt, grid, vars, params, dev)
+end
+
+
+# ----------
+# Parameters
+# ----------
+
+"""
+    Params(ν, nν, μ, nμ, calcF!)
+
+Returns the params for Surface QG turbulence.
+"""
+struct Params{T} <: AbstractParams
+       ν :: T         # Buoyancy viscosity
+      nν :: Int       # Buoyancy hyperviscous order
+       μ :: T         # Bottom drag or hypoviscosity
+      nμ :: Int       # Order of hypoviscosity
+  calcF! :: Function  # Function that calculates the buoyancy forcing F
+end
+
+Params(ν, nν) = Params(ν, nν, typeof(ν)(0), 0, nothingfunction)
+
+
+# ---------
+# Equations
+# ---------
+
+"""
+    Equation(params, grid)
+
+Returns the equation for Surface QG turbulence with params p and `grid`.
+"""
+function Equation(params::Params, grid::AbstractGrid)
+  L = @. - params.ν * grid.Krsq^params.nν - params.μ * grid.Krsq^params.nμ
```

Comment:
This includes hypoviscosity but the docs don't mention anything. Shall we keep hypoviscosity here and add it in the equation in the docs or remove it from here whatsoever?

Reply:
I don't think I have seen an SQG model with hypo-viscosity since energy and buoyancy variance cascades to small scales. I would support removing it.